lib/zipkin-tracer/zipkin_rabbit_mq_sender.rb

```diff
@@ -0,0 +1,42 @@
+require 'json'
+require 'zipkin-tracer/zipkin_sender_base'
+require 'zipkin-tracer/hostname_resolver'
+
+module Trace
+  class RabbitMqPublisher
+    def initialize(connection)
+      @connection = connection
+      @channel = @connection.create_channel
+    end
+
+    def publish(exchange, routing_key, message)
+      exchange = @channel.exchange(exchange)
```

Comment:
Does this operation needs to be done on each `publish`? Both @channel and @exchange does not seem to change.

Reply:
I guess it doesn't really need to be done on each publish, I moved it out.